projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/README.md

```diff
@@ -97,6 +97,42 @@ In development mode, you will also need to remove the symlink created by `jupyte
 command. To find its location, you can run `jupyter labextension list` to figure out where the `labextensions`
 folder is located. Then you can remove the symlink named `vdk-jupyterlab-extension` within that folder.
 
+### Front-end extension
+This extension uses [JSX](https://reactjs.org/docs/introducing-jsx.html).
+
+The components of the front-end extension are located in the /src directory. All the new UI elements are added there.
+
+The main script for the extension is index.ts - this is where the front-end extension is loaded.
+In handlers.ts the connection with the server is done, while in serverRequests.ts all the requests are sent.
+In the subdirectory /components are located the JSX components that represent VDK menu elements.
+In the subdirectory /dataClasses are located the data classes responsible for saving the user input data for all the
+vdk operations.
+
+
+### Server extension
+This extension uses [Tordnado](https://www.tornadoweb.org/en/stable/).
+
+All the requests handlers are located in handlers.py file where the communication with the front-end is created.
+
+The connection with VDK is done in the vdk_ui.py file - all the vdk operations are handled there.
+
+### Job data model
+Follows the enum JobProperties from [ui-job-properties.py](vdk-jupyterlab-extension/ui_job_properties.py).
+
+
+In the front-end extension a global variable is introduces (jobData) which holds the information about the current job.
```

Comment:
so we now use a broader than session-scope (and that is global) variable, for the purposes to solve the UX consistency in multi-tab scenario? Please add a few words about this

Reply:
done